application/src/main/java/org/thingsboard/server/controller/OtaPackageController.java

```diff
@@ -184,32 +167,12 @@ public OtaPackageInfo saveOtaPackageData(@ApiParam(value = OTA_PACKAGE_ID_PARAM_
         checkParameter(CHECKSUM_ALGORITHM, checksumAlgorithmStr);
         try {
             OtaPackageId otaPackageId = new OtaPackageId(toUUID(strOtaPackageId));
-            OtaPackageInfo info = checkOtaPackageInfoId(otaPackageId, Operation.READ);
-
-            OtaPackage otaPackage = new OtaPackage(otaPackageId);
-            otaPackage.setCreatedTime(info.getCreatedTime());
-            otaPackage.setTenantId(getTenantId());
-            otaPackage.setDeviceProfileId(info.getDeviceProfileId());
-            otaPackage.setType(info.getType());
-            otaPackage.setTitle(info.getTitle());
-            otaPackage.setVersion(info.getVersion());
-            otaPackage.setTag(info.getTag());
-            otaPackage.setAdditionalInfo(info.getAdditionalInfo());
-
             ChecksumAlgorithm checksumAlgorithm = ChecksumAlgorithm.valueOf(checksumAlgorithmStr.toUpperCase());
 
-            byte[] bytes = file.getBytes();
             if (StringUtils.isEmpty(checksum)) {
-                checksum = otaPackageService.generateChecksum(checksumAlgorithm, ByteBuffer.wrap(bytes));
+                checksum = ChecksumUtil.generateChecksum(checksumAlgorithm, file.getInputStream());
```

Comment:
This API call will consume the input stream and nothing will be saved to the DB.